common/persistence/visibilitySamplingClient.go

```diff
@@ -176,8 +180,8 @@ func (p *visibilitySamplingClient) ListOpenWorkflowExecutions(
 	domain := request.Domain
 
 	rateLimiter := p.rateLimitersForList.getRateLimiter(domain, numOfPriorityForList, p.config.VisibilityListMaxQPS(domain))
-	if ok, _ := rateLimiter.GetToken(0, 1); !ok {
-		return nil, ErrPersistenceLimitExceededForList
+	if !p.tryConsumeListToken(request.Domain, rateLimiter) {
```

Comment:
Nit/for a consideration:
we can reduce code duplication a bit and keep logic in a single place with a helper method:
```go
func (p *visibilitySamplingClient) isRateLimited(domain string) bool {
	rateLimiter := p.rateLimitersForList.getRateLimiter(domain, numOfPriorityForList, p.config.VisibilityListMaxQPS(domain))
	ok, _ := rateLimiter.GetToken(0, 1)
	if ok {
		p.logger.Debug("List API request consumed QPS token", tag.WorkflowDomainName(domain), tag.Name(callerFuncName(2)))
	}
	return !ok
}
```
Then methods looks simply like this:
```go
	if p.isRateLimited(request.Domain) {
		return nil, p.returnServiceBusyErrorForList(request.Domain)
	}

	return <...>
```

Reply:
that's a good idea. Thanks Mantas. Will do it.

Reply:
Adopted your idea with a lightly difference. :D thanks so much.